### ISSUE TITLE
Failed to install WinRing0 driver, error 1072

### ISSUE THREAD

**Issue by GoGitIt1** [xmrig/xmrig#1423]
I receive this error with the newest version (5.3) of XMRig. Any suggestions on how to fix this?

After relaunching XMRig, I also receive the error "msr failed to stop WinRing0 driver, error 1061" and "msr failed to remove WinRing0 driver, error 1072"
I expect MSR to run, but it looks like it is having a problem. 

Miner log:
![XMRig_Error](https://user-images.githubusercontent.com/58889729/70865145-22219300-1f28-11ea-8ba0-92931bf87482.JPG)

Config file:
start "" "D:\Documents\Cryptocurrency\Monero\Miners\XM_Rig\xmrig.exe" --donate-level 0 -o xmr-us-east1.nanopool.org:14433 -u WALLETADDRESS --tls -k --coin monero  -a rx/0 --max-cpu-usage 95
exit

OS: Windows 8.1

**Comment by xmrig** [comment]
Error 1072 = `ERROR_SERVICE_MARKED_FOR_DELETE`
Error 1061 = `ERROR_SERVICE_CANNOT_ACCEPT_CTRL`

Do you have other application which can use same driver? for hardware monitoring, etc.
Thank you.

**Comment by xmrig** [comment]
This is definitely conflict with other software which use WinRing0 driver. Did you found it? I found one: Open Hardware Monitor.
Thank you.

**Comment by Bathmat** [comment]
I'm also getting errors with MSR. I've tried disabling all my hardware montioring software, but still getting this error:
` msr  failed to start WinRing0 driver, error 2`

**Comment by xmrig** [comment]
@Bathmat You did't place `WinRing0x64.sys` near to `xmrig.exe`.
Thank you.

**Comment by Bathmat** [comment]
Ahh, thanks. Missed extracting that file. Appears to be working now after I closed HW Monitor software

**Comment by xmrig** [comment]
I fixed compatibility with Open Hardware Monitor (error 1072) and probably with other applications which use the driver.
@Bathmat and added better error message for case if the file not found.

**Comment by pawelantczak** [comment]
@GoGitIt1 consider enabling donation since @xmrig is investing his time twice, to develop and to help you.

**Comment by distributorofpain** [comment]
Getting:
"failed to install WinRing0 driver, error 1072" when i start xmrig as admin.

If i kill it and start over, i get the following error:
"failed to stop WinRing0 driver, error 1061"
"failed to remove WinRing0 driver, error 1072"

The computers are windows 10 1903 with only basic software loaded for mining.  I downloaded 5.3.0 just now.  I also made sure the compatibility was set to unblock.

Two Rigs, exactly the same:
AMD 3700X with 16GB 3600 on B450 board.

I have another older machine that i setup as a miner with i7-3770K, most of the same software loaded, but no LED or overclocking software.  That one loads the MSR no issues.

*** I removed RGB Fusion and it fixed the driver load issue.  Seems the RGB software out there uses this driver.  ( i dont need it anyways because it wasnt disabling the RGB on the CPU like i wanted ).  Had to reboot after uninstall.

**Comment by SChernykh** [comment]
@distributorofpain Error 1072 is fixed in dev branch and the fix will be in the next release.

**Comment by xmrig** [comment]
Extra addition, next version will will also show path to exists driver (it helps to find another software witch use same driver) if set `"verbose": 1,` in config file (it also new option).

But it just for information, the miner will reuse already running driver, all of this implemented in dev branch.
Thank you.

**Comment by FAB1150** [comment]
how can I download the latest dev release? I don't use github very often, sorry.

**Comment by pawelantczak** [comment]
Apparently you don't use your reading capability very often too. https://github.com/xmrig/xmrig/wiki/Build
Sorry for sarcasm, but such lack of self-sufficiency and independence can be really annoying.

**Comment by xmrig** [comment]
@FAB1150 https://github.com/xmrig/xmrig/issues/1425#issuecomment-566664690 binaries for every commit created by https://buildbot.xmrig.com/
Thank you.

**Comment by FAB1150** [comment]
@xmrig thank you! I downloaded dev 5.4.0, and i still have error 1072 but now it says "msr  service WinRing0_1_2_0 is already exists" "msr  failed to remove WinRing0 driver, error 1072".
i'll attach a screenshot.
I have HWinfo installed but NOT running, i also rebooted my pc to make sure.

![Screenshot (16)](https://user-images.githubusercontent.com/23463665/71043494-0c31ef00-212f-11ea-8a7d-88becb1c26d6.png)

**Comment by GoGitIt1** [comment]
> 
> 
> Error 1072 = `ERROR_SERVICE_MARKED_FOR_DELETE`
> Error 1061 = `ERROR_SERVICE_CANNOT_ACCEPT_CTRL`
> 
> Do you have other application which can use same driver? for hardware monitoring, etc.
> Thank you.

Thanks for fixing this, XMrig. I turned debugging mode on in Windows to get this in v5.2.1. I think this was the issue:
https://www.reddit.com/r/MoneroMining/comments/e9tuvd/randomx_boost_guide_for_ryzen_on_windows_9100_hs/

I tried disabling every service/app that I could think of, but problem mostly went away after doing this. For whatever reason, MSR was able to be active 2 out of around 20 times with debugging mode on. Not sure why. Glad problem fixed.
